### ISSUE TITLE
`WRONG_INDENTATION` rule behaving incorrectly with superclass constructor call arguments

### ISSUE THREAD

**Issue by 0x6675636b796f75676974687562** [saveourtool/diktat#1404]
Both this

```kotlin
class Klass : Base("foo")
```

and this superclass constructor calls are indented correctly, in full accordance with [3.3. Indentation](https://github.com/saveourtool/diktat/blob/master/info/guide/diktat-coding-convention.md#-33-indentation):

```kotlin
class Klass :
    Base("foo")
```

Now consider we want to split the superclass constructor call arguments into multiple lines:

```kotlin
class Klass : Base(
    "foo"
)
```

Even this is correct, too. Now, let's put a newline _before_ the superclass call:

```kotlin
class Klass :
    Base(
        "foo"
    )
```

_DiKTat_ will immediately complain:

```
Klass.kt:3:1: [WRONG_INDENTATION] only spaces are allowed for indentation and each indentation should equal to 4 spaces (tabs are not allowed): expected 4 but was 8 (diktat-ruleset:indentation)
Klass.kt:4:1: [WRONG_INDENTATION] only spaces are allowed for indentation and each indentation should equal to 4 spaces (tabs are not allowed): expected 0 but was 4 (diktat-ruleset:indentation)
```

So apparently _DiKTat_ expects the following formatting (outright wrong):

```kotlin
class Klass :
    Base(
    "foo"
)
```

## Steps to Reproduce

As it's obvious from the above example, two conditions must be met for the issue to manifest itself:

1. The supertype constructor call should be put on a separate line.
1. The supertype constructor call arguments should be put on individual lines.

There're more real-life examples which trigger this issue, e. g.:

```kotlin
class Klass :
    Base(
        "foo",
        ::Klass,
        listOf(
            "foo",
            "bar",
            "baz"
        )
    ),
    Cloneable,
    CharSequence {
    // ...
}
```

or even

```kotlin
class IndentationRuleFixTest :
    FixTestBase(
        "test/paragraph3/indentation",
        ::IndentationRule,
        listOf(
            RulesConfig(WRONG_INDENTATION.name, true,
                mapOf(
                    "newlineAtEnd" to "true",
                    "extendedIndentOfParameters" to "true",
                    "alignedParameters" to "true",
                    "extendedIndentAfterOperators" to "true",
                    "extendedIndentBeforeDot" to "true",
                )
            )
        )
    ),
    IndentationRuleTestMixin,
    IndentationRuleTestResources {
    // ...
}
```

## Environment information
* diktat version: 1.2.0
* build tool (maven/gradle): Maven
* how is diktat run (CLI, plugin, etc.): Maven plug-in
* kotlin version: 1.7.0
* operating system: Windows

**Comment by akuleshov7** [comment]
Important, let's do it! But not break anything else :)

**Comment by 0x6675636b796f75676974687562** [comment]
More examples:

```kotlin
class C(private val property: (rulesConfigList: List<RulesConfig>) -> Rule,
        arg1: List<RulesConfig>?) :
    Base(
        id = DIKTAT_RULE_SET_ID,
        about = NO_ABOUT,
    ) {
}
```

and

```kotlin
class C
@JvmOverloads
constructor(private val property: DiktatRuleSetFactory = DiktatRuleSetFactory()) :
    Base(
        id = DIKTAT_RULE_SET_ID,
        about = About(
            maintainer = "Diktat",
            description = "Strict coding standard for Kotlin and a custom set of rules for detecting code smells, code style issues, and bugs",
            license = "https://github.com/saveourtool/diktat/blob/master/LICENSE",
            repositoryUrl = "https://github.com/saveourtool/diktat",
            issueTrackerUrl = "https://github.com/saveourtool/diktat/issues",
        ),
    ) {
}
```
